### PR TITLE
Snowflake test check connection basic exceptions

### PR DESCRIPTION
**Pull request by Ismail-Naziz**
## What

Adding test of basic checks: 

- invalid config
- invalid private key
- invalid host name
- duplicated push down filter name
- inconsistent parent stream name
- invalid update method
- unknown schema

Handling error 401 in check connection 

The status is set to failed for error cases

In this folder: integration_tests/resources, you will find configs that you can set for test with dummy values for secrets.

### CODE REVIEW

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py [hunk 1]
```diff
@@ -49,45 +49,35 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
...
             error_message = error.__str__()
             error_code = error.response.status_code
+
             if int(error_code) == 412:
                 error_message = ("SQL execution error for check.\n"
                                  "The origin of the error is very likely to be:\n"
                                  "- The configuration provided does not have enough permissions to access the requested database/schema.\n"
                                  "- The configuration is not consistent (example: schema not present is database.")
+
+            if int(error_code) == 404:
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py:78]: It seems very confusing for the person debugging that the HTTP status is `404` but the error message mentions `401`. What is the reason for that?

**Review by maxi297 — COMMENTED** [review]
I would like to understand more about the error handle for 404 before approving

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py (hunk 1 above)
Comment (maxi297) [reply airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py]: Based on the commit message, it was a typo. How was the test for `secrets/invalid_private_key_config.json` passing then?

Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py]: It was caught by another custom error class (BadPrivateKeyFormatError) I introduced, the error is about deserialization not about having incorrect credential values

Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py]: It is confusing, i was a type from me i corrected it.

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py [hunk 2]
```diff
@@ -35,59 +36,53 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
...
+            authenticator = SnowflakeJwtAuthenticator.from_config(config)
             self.check_update_method(config=config)
             self.check_push_down_filter_name_unicity(config=config)
             url_base = self.format_url_base(host)
-            authenticator = SnowflakeJwtAuthenticator.from_config(config)
 
             self.check_existence_of_at_least_one_stream(url_base=url_base, config=config, authenticator=authenticator)
             self.check_push_down_filters_parent_stream_consistency(url_base=url_base, config=config, authenticator=authenticator)
 
         except IncorrectHostFormat:
```
[the remark this answers is not in the document]
Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py:51]: it is not just about some regex (starts or ends with for example), It is also about the decoding of the private key. This error (BadPrivateKeyFormatError) is raised twice for 2 different reasons in the code, but for one reason for the user : he did not use the PEM format.

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py [hunk 3]
```diff
@@ -35,59 +36,53 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
...
             error_code = error.response.status_code
+
             if int(error_code) == 412:
                 error_message = ("SQL execution error for check.\n"
                                  "The origin of the error is very likely to be:\n"
                                  "- The configuration provided does not have enough permissions to access the requested database/schema.\n"
                                  "- The configuration is not consistent (example: schema not present is database.")
+
+            if int(error_code) == 401:
+                error_message = "401 Client Error: Unauthorized for url. Make sure you are using the correct credentials"
```
[the remark this answers is not in the document]
Comment (Ismail-Naziz) [reply airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py:84]: Done

File: airbyte-integrations/connectors/source-snowflake/integration_tests/resources/incorrect_host_format_config.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        "private_key": "-----BEGIN PRIVATE KEY----- your key content -----END PRIVATE KEY-----"
+    },
+    "streams": [],
+    "host": "https://XXXXXXXXXXX.not_right_domain.com",
+    "role": "ROLE",
+    "warehouse": "WAREHOUSE",
+    "database": "DATABASE",
+    "schema": "SCHEMA",
+    "replication_method":{"method":"standard"}
+}
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-snowflake/integration_tests/resources/incorrect_host_format_config.json:14]: nit: extra line is needed.

File: airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py [hunk 5]
```diff
@@ -15,7 +15,8 @@
 from airbyte_cdk.models import Type as AirbyteType
 from .authenticator import SnowflakeJwtAuthenticator
 from .snowflake_exceptions import InconsistentPushDownFilterParentStreamNameError, NotEnabledChangeTrackingOptionError, \
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py:17]: nit:
```python
from .snowflake_exceptions import (
    InconsistentPushDownFilterParentStreamNameError, NotEnabledChangeTrackingOptionError,
    DuplicatedPushDownFilterStreamNameError,
    IncorrectHostFormat, 
    emit_airbyte_error_message, 
    UnknownUpdateMethodError,
    BadPrivateKeyFormatError,
)
```

**Review by bazarnov — COMMENTED** [review]
Looks good!

**Pull request merged**